WalletWasabi/Helpers/HashHelpers.cs

```diff
@@ -21,5 +21,29 @@ public static byte[] GenerateSha256Hash(byte[] input)
 
 			return hash;
 		}
+
+		/// <summary>
+		/// https://stackoverflow.com/a/468084/2061103
+		/// </summary>
+		public static int ComputeHashCode(params byte[] data)
+		{
+			unchecked
+			{
+				const int P = 16777619;
+				int hash = (int)2166136261;
+
+				for (int i = 0; i < data.Length; i++)
+				{
+					hash = (hash ^ data[i]) * P;
+				}
+
+				hash += hash << 13;
+				hash ^= hash >> 7;
+				hash += hash << 3;
+				hash ^= hash >> 17;
+				hash += hash << 5;
+				return hash;
+			}
+		}
```

Comment:
I think the SO answer is old. Please consider using the new `HashCode` class for all hash code related calculations:

```suggestion
		public static int ComputeHashCode(params byte[] data)
		{
			var hash = new HashCode();

			foreach (var element in data)
			{
				hash.Add(element);
			}
			return hash.ToHashCode();
		}
```